ibis/backends/tests/test_client.py

```diff
@@ -1629,3 +1629,34 @@ def test_from_connection(con, top_level):
     new_con = backend.from_connection(getattr(con, CON_ATTR.get(con.name, "con")))
     result = int(new_con.execute(ibis.literal(1, type="int")))
     assert result == 1
+
+
+def test_no_accidental_cross_database_table_load(con_create_database):
```

Comment:
It's amazing that you were able to write a cross backend test for this.

I have a test case that I will push up for DuckDB that involves `attach`, to test the robustness of the implementation across catalogs that also contain an overlapping table name.